flytepropeller/pkg/controller/nodes/attr_path_resolver.go

```diff
@@ -97,6 +105,79 @@ func resolveAttrPathInPbStruct(nodeID string, st *structpb.Struct, bindAttrPath
 	return literal, err
 }
 
+// resolveAttrPathInBinary resolves the binary idl object (e.g. dataclass, pydantic basemodel) with attribute path
+func resolveAttrPathInBinary(nodeID string, binaryIDL *core.Binary, bindAttrPath []*core.PromiseAttribute) (*core.
+	Literal,
+	error) {
+
+	binaryBytes := binaryIDL.GetValue()
+	serializationFormat := binaryIDL.GetTag()
+
+	var currVal interface{}
+	var tmpVal interface{}
+	var exist bool
+
+	if serializationFormat == messagepack {
+		err := msgpack.Unmarshal(binaryBytes, &currVal)
+		if err != nil {
+			return nil, err
+
+		}
+	} else {
+		return nil, errors.Errorf(errors.PromiseAttributeResolveError, nodeID,
+			"Unsupported format '%v' found for literal value.\n"+
+				"Please ensure the serialization format is supported.", serializationFormat)
+	}
+
+	// Turn the current value to a map, so it can be resolved more easily
+	for _, attr := range bindAttrPath {
+		switch resolvedVal := currVal.(type) {
+		// map
+		case map[interface{}]interface{}:
+			tmpVal, exist = resolvedVal[attr.GetStringValue()]
```

Comment:
What will GetStringValue() return if if the attr is `foo[0]`? shouldn't it fail in that case and say "you can't index into a struct" or something ?

Reply:
Yes, just fix this case here.
https://github.com/flyteorg/flyte/pull/5763/files#diff-ee7f936e440a7e043b3bc7acb4ea255ba991dea8f3144d24ab276c3a292de018R136-R164